lib/promise.js

```diff
@@ -1,33 +1,43 @@
 'use strict';
 
 Promise.each =  async function(arr, fn) {
-   for(const item of arr) await fn(item);
+    for(const item of arr) await fn(item);
 }
 
-Promise.prototype.return = func => {
-  return Promise.resolve(func);
+Promise.prototype.return = function(value) {
+  return this.then(function(){ return value });
 }
 
 Promise.prototype.tap = function(fn) {
-    return this.then(function(value){
-         fn(value);
+    return this.then(async function(value){
```

Comment:
Let's not use async here in terms of performance.

Reply:
Don't have a choice trust me I wanted to.

Reply:
Just curious, why can't you return a Promise if async/await is already doing this?